src/test/java/org/minions/devfund/yuri/WackAMoleTest.java

```diff
@@ -0,0 +1,229 @@
+package org.minions.devfund.yuri;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+/**
+ * It is in charge to test the WackAMole class functionality.
+ */
+public class WackAMoleTest {
+    public static final int POS_X = 1;
+    public static final int POS_Y = 3;
+    public static final int GRID_DIMENSION = 10;
+    public static final int MOLES_AMOUNT = 10;
+    public static final int CELLS_AMOUNT = 100;
+    public static final int NUM_ATTEMPTS = 50;
+    public static final char MOLE = 'M';
+    public static final char WACKED = 'W';
+    public static final char EMPTYCELL = '*';
+    private WhackAMole game;
+
+    /**
+     * Method setup.
+     */
+    @Before
+    public void setUp() {
+        game = new WhackAMole(NUM_ATTEMPTS, GRID_DIMENSION);
+        game.insertMoles(GRID_DIMENSION);
+    }
+
+    /**
+     * Verifies that a mole is placed in specific cell.
+     */
+    @Test
+    public void testMoleIsPlacedInSpecificCell() {
+        game.place(POS_X, POS_Y);
+        assertEquals(MOLE, game.getMoleGrid()[POS_X][POS_Y]);
+    }
+
+    /**
+     * Verifies that an specific cell is empty.
+     */
+    @Test
+    public void testAnSpecificCellIsEmpty() {
+        game.getMoleGrid()[POS_X][POS_Y] = EMPTYCELL;
+        assertTrue(game.isEmptyCell(POS_X, POS_Y));
+    }
+
+    /**
+     * Verifies that an specific cell is a mole.
+     */
+    @Test
+    public void testAnSpecificCellIsMole() {
+        game.getMoleGrid()[POS_X][POS_Y] = MOLE;
+        assertTrue(game.isMole(POS_X, POS_Y));
+    }
+
+    /**
+     * Verifies the number attempts configured is correct.
+     */
+    @Test
+    public void testNumberAttemptsIsCorrect() {
+        assertEquals(NUM_ATTEMPTS, game.getAttemptsLeft());
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void testMolesAmountInsertedInGridIsCorrect() {
+        game.insertMoles(MOLES_AMOUNT);
+        int accumulator = 0;
+        for (int row = 0; row < game.getDimension(); row++) {
+            for (int column = 0; column < game.getMoleGrid()[row].length; column++) {
+                if (game.getMoleGrid()[row][column] == MOLE) {
+                    accumulator++;
+                }
+            }
+        }
+        assertEquals(MOLES_AMOUNT, accumulator);
+    }
+
+    /**
+     * Verifies that a wack is inserted in the grid.
+     */
+    @Test
+    public void testWackInsertedInGridIsCorrect() {
+        game.whack(POS_X, POS_Y);
+        assertEquals(WACKED, game.getMoleGrid()[POS_X][POS_Y]);
+    }
+
+    /**
+     * Verifies that the moles left decrease when a moles is hit.
+     */
+    @Test
+    public void testMolesLeftDecreaseWhenAMoleIsWhacked() {
+        final int attempts = 5;
+        final int dimension = 10;
+        WhackAMole game = new WhackAMole(attempts, dimension);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "game" which hides the field declared at line 24. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)